frontend/src/scenes/session-recordings/player/rrweb/canvas/canvas-plugin.ts

```diff
@@ -65,44 +205,32 @@ export const CanvasReplayerPlugin = (events: eventWithTime[]): ReplayPlugin => {
         },
 
         // eslint-disable-next-line @typescript-eslint/no-misused-promises
-        handler: async (e: eventWithTime, _isSync: boolean, { replayer }: { replayer: Replayer }) => {
-            if (e.type === EventType.IncrementalSnapshot && e.data.source === IncrementalSource.CanvasMutation) {
-                const source = replayer.getMirror().getNode(e.data.id) as HTMLCanvasElement
-                const target = canvases.get(e.data.id) || (source && cloneCanvas(e.data.id, source))
+        handler: async (e: eventWithTime, isSync: boolean, { replayer }: { replayer: Replayer }) => {
+            const isCanvas = isCanvasMutation(e)
 
-                if (!target) {
-                    return
-                }
+            // scrubbing / fast forwarding
+            if (isSync) {
+                // reset preload index
+                nextPreloadIndex = null
+                canvasEventMap.clear()
 
-                if (source) {
-                    target.width = source.clientWidth
-                    target.height = source.clientHeight
-                }
-
-                await canvasMutation({
-                    event: e,
-                    mutation: e.data,
-                    target: target,
-                    imageMap,
-                    canvasEventMap,
-                    errorHandler: () => {},
-                })
-
-                const img = containers.get(e.data.id)
-                if (img) {
-                    target.toBlob((blob) => {
-                        if (blob) {
-                            img.style.width = 'initial'
-                            img.style.height = 'initial'
-
-                            const url = URL.createObjectURL(blob)
-                            // no longer need to read the blob so it's revoked
-                            img.onload = () => URL.revokeObjectURL(url)
-                            img.src = url
-                        }
-                    })
+                if (isCanvas) {
+                    processMutationSync(e, { replayer })
+                } else {
+                    pruneBuffer(e)
                 }
+                pruneBuffer(e)
+            } else if (isCanvas) {
+                void processMutation(e, replayer).catch(handleMutationError)
             }
         },
     } as ReplayPlugin
 }
+
+const handleMutationError = (error: unknown): void => {
+    if (error instanceof Error) {
+        captureException(error)
+    } else {
+        console.error(error)
```

Comment:
What else can be thrown besides an `Error`?
If a string, we could either `captureMessage` or create a synthetic error? Error(error) so we can debug if things go south.

Reply:
I think it was more for typing given the catch was throwing an `unknown` type. Sentry actually accepts `any` so we should just use that instead